### PR TITLE
Update logr dependency (#122)

### PR DESCRIPTION
**Pull request by solonish**
* Upgraded logr version to v1.2.2, upgraded sigs.k8s.io/controller-runtime to v0.11.0 - https://github.com/kubernetes-sigs/controller-runtime/issues/1607, upgraded cert-manager version to v1.7.1

Signed-off-by: Nishanth Solomon <solonish@amazon.com>

* Changed NullLogger to Discard - NullLogger is deprecated https://pkg.go.dev/github.com/go-logr/logr@v0.4.0/testing#NullLogger

Signed-off-by: Nishanth Solomon <solonish@amazon.com>

* Changed TestLogger to NewTestLogger - https://github.com/go-logr/logr/releases/tag/v1.1.0

Signed-off-by: Nishanth Solomon <solonish@amazon.com>

* Updated to golang:1.17 - https://github.com/kubernetes-sigs/controller-tools/issues/643

Signed-off-by: Nishanth Solomon <solonish@amazon.com>

### CODE REVIEW

File: pkg/aws/pca_test.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ import (
 	"github.com/aws/aws-sdk-go-v2/service/acmpca/types"
 	acmpcatypes "github.com/aws/aws-sdk-go-v2/service/acmpca/types"
 
-	testLog "github.com/go-logr/logr/testing"
+	testLog "github.com/go-logr/logr"
```
Comment (bmsiegel) [inline pkg/aws/pca_test.go:35]: Can we change this to `logr`? It's not the testing package from `logr` anymore, and that way it matches `logrtesting` from the other file

**Comment by bmsiegel** [comment]
Let's add a README update to change our go minimum version to 1.17 as well

File: pkg/aws/pca_test.go (hunk 1 above)
Comment (solonish) [reply pkg/aws/pca_test.go]: Removed testLog

File: go.mod [hunk 2]
```diff
@@ -3,21 +3,19 @@ module github.com/cert-manager/aws-privateca-issuer
 go 1.15
```
Comment (bmsiegel) [inline go.mod:3]: Change this to 1.17

Comment (solonish) [reply go.mod]: Changed to 1.17

**Pull request merged**